internal/kafka/internal/workers/clusters_mgr.go

```diff
@@ -561,6 +593,27 @@ func (c *ClusterManager) reconcileClusterDNS(cluster api.Cluster) error {
 	return nil
 }
 
+func (c *ClusterManager) reconcileKasFleetshardOperator(cluster api.Cluster) error {
+	if c.KasFleetshardOperatorAddon != nil {
```

Comment:
`KasFleetshardOperatorAddon` should always be set, I am wondering why we are checking for this `nil` check. I suspect it because of the test. 

It is an existing issue and we do not have to investigate and address the issue on this PR, but definitely something we should look into addressing after this PR.

Reply:
Thanks for pointing this out. I will look into it :)